.github/workflows/test-and-upload-coverage.yml

```diff
@@ -27,19 +27,26 @@ jobs:
   run-tests:
     name: Run tests matrix job
 
-    runs-on: ubuntu-latest
-
     strategy:
       matrix:
```

Comment:
thought(out of scope): While you are at it, perhaps an explicit `fail-fast: true` (which is the default behavior at the moment). To have it stay consistent incase GitHub decides to change the default in the future.